brickstrap.sh

```diff
@@ -120,96 +146,172 @@ function help() {
 ### are simply ignored
 #####################################################################
 
-BOARDDIR="ev3-ev3dev-jessie"
-
-while [ $# -gt 0 ] ; do
-    while getopts "fb:d:" opt; do
-        case "$opt" in
-           f) FORCE=true;;
-           b) BOARDDIR="$OPTARG";;
-           d) ROOTDIR="$OPTARG";;
-          \?) # unknown flag
-                help
+function brp_parse_cli_options()
+{
+    while [ $# -gt 0 ] ; do
+        while getopts "fhb:c:d:p:" BRP_OPT; do
+            case "$BRP_OPT" in
+            f) FORCE=true;;
+            h)
+                brp_help
+                exit 0
+            ;;
+            b|p)
+                if [ -z "$OPTARG" ]; then
+                    brp_help 'Empty project names are invalid'
+                    exit 1
+                elif [ -z "$BR_PROJECT" ]; then
+                    BR_PROJECT="$OPTARG"
+                else
+                    brp_help "Duplicate project: -$BRP_OPT '$OPTARG'.
+Project was: '$BR_PROJECT'"
+                    exit 1
+                fi
+            ;;
+            c)
+                if [ -z "$OPTARG" ]; then
+                    brp_help 'Empty component names are invalid'
+                    exit 1
+                elif [ -z "$BR_COMPONENTS" ]; then
+                    BR_COMPONENTS="'$OPTARG'"
+                else
+                    BR_COMPONENTS="$BR_COMPONENTS '$OPTARG'"
+                fi
+            ;;
+            d) ROOTDIR="$OPTARG";;
+            \?) # unknown flag
+                brp_help "Invalid option specified"
                 exit 1
             ;;
-        esac
-    done
-
-    shift $((OPTIND-1))
-
-    if [ "${cmd}" == "run-hook" ]; then
-      run_hook_arg=$1
-    fi
+            \:) # missing argument
+                brp_help "Missing argument for option: -$BRP_OPT"
```

Comment:
Actually, no it doesn't: turns out this is dead code. I added this based on the getopt C API which uses this convention, but turns out that bash getopt simply prints:
> ../brickstrap/brickstrap.sh: option requires an argument -- p
Invalid option specified

Followed by the usage

Meanwhile an illegal option like `-Z` gets you:

> ../brickstrap/brickstrap.sh: illegal option -- Z
Invalid option specified

That means "Invalid option specified" is also null-content. I'll update the PR to correct this and remove dead-code/null-content.

Note that it is explicitly possible to pass empty things using quotes `brickstrap.sh -p ""`. That's what the checks for empty names are about.